model.py

```diff
@@ -339,7 +345,7 @@ def search_by_location_name(cls, _db, query, type=None, here=None):
         # For a library to match, the Place named by the query must
         # intersect a Place served by that library.
         named_place = aliased(Place)
-        qu = _db.query(Library).join(
+        qu = _db.query(Library).distinct().join(
```

Comment:
Probably? I think it will find the shorter distance first, because of the query ordering and then ignore the library when it shows up again with a longer distance. But even if that's wrong, the distances will be comparable in most cases, because they'll be adjacent cities or different parts of a single city.